### PR TITLE
feat: fix alter suggestions; fix table and column suggestions with multiple statements; add alter, drop tests; separate newline tests; add commit and ci checks

### PR DESCRIPTION
**Pull request by roberthovsepyan**

### CODE REVIEW

File: src/tests/shared/newline.test.ts [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        line: 2,
+        column: 1,
+    });
+    const collumnSuggestions: ColumnSuggestion = {
+        tables: [{name: 'test_table'}],
+    };
+
+    parseResults.forEach(({suggestColumns}) => {
+        expect(suggestColumns).toEqual(collumnSuggestions);
+    });
```
Comment (NikitaShkaruba) [inline src/tests/shared/newline.test.ts:15]: Shouldn't we check for errors here? It's a more simpler test that will not break if your column suggestions code will break. Same for all the other test cases

**Review by NikitaShkaruba — APPROVED** [review]

File: src/tests/shared/newline.test.ts (hunk 1 above)
Comment (roberthovsepyan) [reply src/tests/shared/newline.test.ts:15]: checking for errors will only test if grammar is correct
and these tests check that our code handles newlines correctly

but I can add tests for errors as well

File: src/tests/shared/alter/drop-column.test.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+test('should not report errors on multiple statements', () => {
+    const parseResults = groupParseSqlWithoutCursor(`
+      ALTER TABLE test_table DROP COLUMN id;\n
+      ALTER TABLE test_table DROP COLUMN id;
+  `);
+
+    parseResults.forEach(({errors}) => {
+        expect(errors).toHaveLength(0);
+    });
+});
```
Comment (NikitaShkaruba) [inline src/tests/shared/alter/drop-column.test.ts:29]: Why do we need to check if it works for multiple statements? We should have only one multi-statement test for it, which should not include columns in itself

File: src/tests/shared/newline.test.ts (hunk 1 above)
Comment (NikitaShkaruba) [reply src/tests/shared/newline.test.ts:15]: I meant that we want from our newline test is to test that it works with newlines. It should be super simple. Something like;
`SELECT 1;\nSELECT1` -> check that there are no errors

File: src/tests/shared/alter/rename-column.test.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        `
+      ALTER TABLE test_table RENAME COLUMN id TO name;\n
+      ALTER TABLE test_table RENAME COLUMN id TO name;
+    `,
+    );
+
+    parseResults.forEach(({errors}) => {
+        expect(errors).toHaveLength(0);
+    });
+});
```
Comment (NikitaShkaruba) [inline src/tests/shared/alter/rename-column.test.ts]: We don't need to check if multiple statements work. We should have a separate test for it.

Here we'd like to check that a correct table name will be suggested if there are multiple table rules;
```
ALTER TABLE test_table_1 RENAME COLUMN id;
ALTER TABLE test_table_2 RENAME COLUMN id |
ALTER TABLE test_table_3 RENAME COLUMN id;
```

We need to check that only columns from test_table_2 are being suggested

Comment (NikitaShkaruba) [reply src/tests/shared/alter/rename-column.test.ts]: Same for all the tests that try to find other rules or tokens via `preferredRules` or `Visitor`. If it doesn't work, we should mark it as `test.skip`. It should be fixed everywhere.

Also if a statement can be included in the other one, we should create a separate test for it too.

File: src/tests/clickhouse/drop/drop.test.ts [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    const keywords: KeywordSuggestion[] = [{value: 'TABLE'}];
+
+    expect(parseResult.suggestKeywords).toEqual(keywords);
+});
+
+test('should suggest tables after DROP TEMPORARY TABLE', () => {
+    const parseResult = parseClickHouseQueryWithCursor('DROP TEMPORARY TABLE |');
+
+    expect(parseResult.suggestViewsOrTables).toEqual(TableOrViewSuggestion.TABLES);
+});
```
Comment (NikitaShkaruba) [inline src/tests/clickhouse/drop/drop.test.ts:28]: You should check if `DROP VIEW asdf; DROP TABLE adsf;` will suggest tables for you. If it doesn't work, you can skip this test. But we should have it.

File: src/tests/shared/newline.test.ts (hunk 1 above)
Comment (NikitaShkaruba) [reply src/tests/shared/newline.test.ts]: I feel like it's an overkill. Yes, we can test whatever we want, but it should be kinda obvious that we should use lexer to suggest things, not regular expressions or tokenStream. And lexer handles spaces and newlines wrapping automatically. So we need to only have one test for it.

File: src/index.ts [hunk 5]
```diff
@@ -74,7 +74,7 @@ function getColumnSuggestions<
     getParseTree: GetParseTree<P>,
     initialTokenStream: TokenStream,
     cursor: CursorPosition,
-    currentStatement: string,
+    initQuery: string,
```
Comment (NikitaShkaruba) [inline src/index.ts:77]: nit: we should name it `initialQuery`. We don't shorten variable names

File: src/tests/shared/alter/alter-column.test.ts [hunk 6]
```diff
@@ -12,6 +12,18 @@ test('should suggest table name after ALTER COLUMN', () => {
     });
 });
 
+test('should suggest table name after ALTER COLUMN between statements', () => {
+    const parseResults = groupParseSqlWithCursor(
+        'ALTER TABLE before_table DROP COLUMN id; ALTER TABLE test_table ALTER COLUMN | ; ALTER TABLE after_table DROP COLUMN id;',
+        [DatabaseType.MySql, DatabaseType.PostgreSql],
```
Comment (NikitaShkaruba) [inline src/tests/shared/alter/alter-column.test.ts:18]: Why don't we include clickhouse here? Maybe we should just comment out the not working database with a comment like `clickhouse has no alter table statement`?

Comment (roberthovsepyan) [reply src/tests/shared/alter/alter-column.test.ts:18]: clickhouse doesn't have `ALTER COLUMN` clause
I only leave out databases when something is not supported in that database
If something is simply not working I always separate the test and leave a comment

Comment (NikitaShkaruba) [reply src/tests/shared/alter/alter-column.test.ts:18]: yeah, it's good, but from the outside point of view, I can't understand why clickhouse is omitted, maybe it's a bug? But comment would explain it. Anyway, I think we can leave it as is and fix it later if it makes readability worse

**Review by NikitaShkaruba — APPROVED** [review]

**Pull request merged**